### PR TITLE
Ch 452: Information architecture changes

### PR DESCRIPTION
**Pull request by eshtadc**

### CODE REVIEW

File: personalize.admin.inc [hunk 1]
```diff
@@ -249,6 +250,158 @@ function personalize_agent_list() {
...
+  $rows = array();
+  $filter = isset($form_state['values']['filter']) ? $form_state['values']['filter'] : '';
+  foreach($option_sets as $option_set) {
+    $types[$option_set->plugin] = $option_set->plugin;
+    if (!empty($filter) && $filter != $option_set->plugin) {
+      continue;
+    }
+    $option_set_plugin = personalize_get_option_set_type($option_set->plugin);
+    $edit_link = module_invoke($option_set_plugin['module'], 'personalize_edit_link', $option_set);
+    $delete_link = module_invoke($option_set_plugin['module'], 'personalize_delete_link', $option_set);
```
Comment (katbailey) [inline personalize.admin.inc:276]: Should maybe handle the possibility that the module in question hasn't implemented these hooks?

File: personalize.admin.inc [hunk 2]
```diff
@@ -249,6 +250,158 @@ function personalize_agent_list() {
...
+ * Form creation function to delete a specified goal.
+ *
+ * @param array $goal
+ *   The goal to be deleted.
+ */
+function personalize_admin_goal_delete($form, &$form_state, $goal) {
+  $options = personalize_goals_options();
+  $campaign = personalize_agent_load($goal['agent']);
+  if ($campaign->status == PERSONALIZE_STATUS_RUNNING) {
+    $description = t('The %goal goal will be no longer be tracked in the currently running %campaign campaign.', array(
```
Comment (katbailey) [inline personalize.admin.inc:375]: '...will be no longer be...'

File: personalize.admin.inc [hunk 3]
```diff
@@ -249,6 +250,158 @@ function personalize_agent_list() {
...
+      '%campaign' => $campaign->label,
+      '%goal' => $options[$goal['action']]
+    ));
+  }
+  else {
+    $description = t('The %goal goal will be no longer be tracked when the %campaign campaign is started.', array(
```
Comment (katbailey) [inline personalize.admin.inc:381]: same :)

Comment (eshtadc) [inline personalize.admin.inc:381]: love copy/paste

**Comment by eshtadc** [comment]
Merged manually.

**Pull request closed without merging**